### PR TITLE
[FEAT] Added checks and plots based on oxygen SOP

### PR DESCRIPTION
**Pull request by MOchiara**

### CODE REVIEW

**Comment by MOchiara** [comment]
Currently the spike function is the least efficient thing ever produced so any suggestions are more than welcome

**Comment by callumrollo** [comment]
I think this might be a case where it is best to use existing libraries rather than reinventing the wheel. In this case, I would recommend looking at [ioos_qc](https://github.com/ioos/ioos_qc) which implements the QARTOD toolbox including global range, spike test and stuck values.

I wrote an example notebook with some glider data for it https://github.com/ioos/ioos_qc/blob/main/docs/source/examples/QartodTestExample_Glider.ipynb

**Comment by MOchiara** [comment]
Uhh fair point! I will check it now and implement it, thanks for the suggestion!

**Comment by callumrollo** [comment]
It might be something to discuss with later today, OG1 support a standardised approach to QC, so we could have a function called pretty early on that runs QC on the whole dataset, then we can just pass that dataset with QC vars to various plotting functions

**Comment by MOchiara** [comment]
image

File: requirements.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ skyfield
 gsw
 sgp4>=2.14
 cartopy
```
Comment (callumrollo) [inline requirements.txt:12]: ioos_qc should be in here too I think?

File: tests/test_plots.py [hunk 2]
```diff
@@ -3,6 +3,7 @@
 import matplotlib.pyplot as plt
 import numpy as np
 import matplotlib
+import ioos_qc.qartod as ioos_qc
```
Comment (callumrollo) [inline tests/test_plots.py:6]: this is a very confusing way to do an import. How about `from ioos_qc import qartod` ?

**Review by callumrollo — CHANGES_REQUESTED** [review]

**Review by callumrollo — APPROVED** [review]

**Pull request merged**